lib/zipkin-tracer/rack/zipkin_env.rb

```diff
@@ -64,12 +60,21 @@ def sampled_header_value(parent_trace_sampled)
       if parent_trace_sampled  # A service upstream decided this goes in all the way
         parent_trace_sampled
       else
-        new_sampled_header_value(force_sample? || current_trace_sampled? && !filtered?)
+        new_sampled_header_value(force_sample? || current_trace_sampled? && !filtered? && routable_request?)
```

Comment:
> Whitelist plugin ensure a request is traced even if it is not routable

If so, I think this line should be:
```ruby
new_sampled_header_value(current_trace_sampled? && !filtered? && (routable_request? || force_sample?))
```

and you need to update README.md and comments 😄

Reply:
It depends on what we want whitelist. to do, if its only to overwrite the routing or to overwrite any decision including sampling. I think it makes more sense as the later, something like when your request hits `/procesingMoney` then it always samples 100%

Reply:
Is the ordering of `current_trace_sampled? && !filtered? && routable_request?` performance related or could it can be changed to `current_trace_sampled? && routable_request? && !filtered?`

Reply:
you are right, routable_request? is much more expensive so goes last

Reply:
I see no need to update readme, I think it reflects reality already